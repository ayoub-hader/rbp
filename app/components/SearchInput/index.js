/**
 *
 * SearchInput
 *
 */

import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: row;
`;

export default function SearchInput({ placeholder, handleSearch }) {
  const searchRef = useRef();
  const handleClick = () => handleSearch(searchRef.current.value);
  const handleKeyPress = event => {
    if (event.key === 'Enter') {
      handleSearch(searchRef.current.value);
    }
  };
  return (
    <Wrapper>
      <input
        ref={searchRef}
        type="search"
        placeholder={placeholder}
        onKeyPress={handleKeyPress}
      />
      <button type="button" onClick={handleClick}>
        Search
      </button>
      <p>tester font</p>
    </Wrapper>
  );
}

SearchInput.propTypes = {
  placeholder: PropTypes.string,
  handleSearch: PropTypes.func.isRequired,
};
