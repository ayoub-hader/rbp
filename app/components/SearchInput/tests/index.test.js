/* eslint-disable import/first */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import SearchInput from '..';

describe('<SearchInput />', () => {
  let renderResult;
  let handleSearch;
  beforeEach(() => {
    handleSearch = jest.fn();
    renderResult = render(
      <SearchInput
        handleSearch={handleSearch}
        placeholder="search something"
      />,
    );
  });

  it('should render and match snapshot ', () => {
    const {
      container: { firstChild },
    } = renderResult;

    expect(firstChild).toMatchSnapshot();
  });

  it('should call handleSearch with appropriate text', () => {
    const {
      container: { firstChild },
    } = renderResult;

    expect(handleSearch).toHaveBeenCalledTimes(0);
    firstChild.querySelector('input').value = 'jumanji';
    fireEvent.click(firstChild.querySelector('button'));

    expect(handleSearch).toHaveBeenCalledTimes(1);
    expect(handleSearch).toHaveBeenCalledWith('jumanji');
  });
});
