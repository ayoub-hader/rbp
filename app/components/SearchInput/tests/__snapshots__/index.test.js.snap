// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render and match snapshot  1`] = `
<div
  class="SearchInput__Wrapper-sc-1ot3cvd-0 Tcmdk"
>
  <input
    placeholder="search something"
    type="search"
  />
  <button
    type="button"
  >
    Search
  </button>
</div>
`;
